values.yaml

```diff
@@ -1039,6 +1043,146 @@ meshGateway:
+  # of annotations, defining any of these values in the `gateways` list
+  # will override the default values provided here.
+  defaults:
+    # Number of replicas for each ingress gateway defined.
+    replicas: 2
+
+    # What gets registered as WAN address for the gateway which is
+    # used as the address for the `.ingress.consul` DNS lookup.
+    wanAddress:
+      # source configures where to retrieve the WAN address (and possibly port)
+      # for the ingress gateway from.
+      # Can be set to either: Service, NodeIP, NodeName or Static.
+      #
+      # Service - Determine the address based on the service type.
+      #   If service.type=LoadBalancer use the external IP or hostname of
+      #   the service. Use the port set by service.port.
+      #   If service.type=NodePort use the Node IP. The port will be set to
+      #   service.nodePort so service.nodePort cannot be null.
+      #   If service.type=ClusterIP use the ClusterIP. The port will be set to
+      #   service.port.
+      #   service.type=ExternalName is not supported.
+      # NodeIP - The node IP as provided by the Kubernetes downward API.
+      # NodeName - The name of the node as provided by the Kubernetes downward
+      #   API. This is useful if the node names are DNS entries that
+      #   are routable from other datacenters.
+      # Static - Use the address hardcoded in wanAddress.static.
+      source: "Service"
+
+      # Port that gets registered for WAN traffic.
+      # If source is set to "Service" then this setting will have no effect.
+      # See the documentation for source as to which port will be used in that
+      # case.
+      port: 443
+
+      # If source is set to "Static" then this value will be used as the WAN
+      # address of the ingress gateways. This is useful if you've configured a
+      # DNS entry to point to your ingress gateways.
+      static: ""
+
+    # Optional hostPort for the gateway to be exposed on.
+    # This can be used with wanAddress.port and wanAddress.useNodeIP
+    # to expose the gateways directly from the node.
+    # NOTE: Cannot set to 8500 or 8502 because those are reserved for the Consul
+    # agent.
+    hostPort: null
+
+    # The service options configure the Service that fronts the gateway Deployment.
+    service:
+      # Type of service, ex. LoadBalancer, ClusterIP.
+      type: ClusterIP
+
+      # Port that the service will be exposed on. This port will be used in the
+      # Consul dns lookup for the ingress gateway.
+      port: 443
+
+      # Optionally hardcode the nodePort of the service if using type: NodePort.
+      # If not set and using type: NodePort, Kubernetes will automatically assign
+      # a port.
+      nodePort: null
+
+      # Additional ports to expose on both the service and the ingress gateway
+      # container for any listeners that will be defined in the gateway's
+      # configuration entry.
+      additionalPorts: [80]
```

Comment:
What about hostPort and nodePort? The use-cases I'm thinking about:

1. I'm running my ingress gateway on one node and using that node's IP address to route to it through the hostPort instead of using the kubernetes service
1. I want to hard-code all the nodePorts for my ingress gateway service–this one I'm not sure if it will ever be used

Currently there's no way to configure nodePorts and hostPorts.

In the end, we need to configure on the deployment for each listener:
* containerPort
* hostPort

and on the service for each listener:
* port
* targetPort
* nodePort

deploy/containerPort == service/targetPort so we can use a single value for that, but what about hostPort and nodePort?

What about supporting:
```yaml
additionalPorts:
- port: 8443 # sets service/port, service/targetPort, deploy/containerPort
  nodePort: 8443 # sets service/nodePort
  hostPort: 8443 # set deploy/hostPort
```

Or maybe have
```yaml
useHostPorts: true
additionalPorts:
- 8443 # sets service/port, service/targetPort, deploy/containerPort, deploy/hostPort
```

That one won't work for setting nodePorts but not sure if that's a legit use-case?